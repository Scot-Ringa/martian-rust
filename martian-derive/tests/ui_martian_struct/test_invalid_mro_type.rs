use martian_derive::MartianStruct;

struct Foo;

#[derive(MartianStruct)]
struct InvalidField {
    num_reads: i64,
    #[mro_type="foo"] // Invalid type foo
    foo: Foo,
}

fn main() {}
