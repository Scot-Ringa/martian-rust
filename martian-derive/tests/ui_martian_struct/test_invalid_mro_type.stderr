error: "The usage of mro_type should be of form #[mro_type="type"] or #[mro_type="type[]"], where type can be one of: int, float, string, bool, map, path". Found foo
 --> $DIR/test_invalid_mro_type.rs:8:5
  |
8 | /     #[mro_type="foo"] // Invalid type foo
9 | |     foo: Foo,
  | |____________^
